pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -53,6 +53,7 @@ type BindMechanism interface {
 	discoverPodNetworkInterface() error
 	preparePodNetworkInterface() error
 	generateDomainIfaceSpec() api.Interface
+	generateDhcpConfig() *cache.DhcpConfig
```

Comment:
The  DhcpConfig is always generated the same way, we can just remove this and add it to podnic

```golang
        dhcpConfig := &cache.DhcpConfig{Name: l.podInterfaceName}               
        if mac != nil {                                                         
            dhcpConfig.MAC = *mac                                               
        }     
```

Reply:
Nope. That member of the `BindMechanism` implementations is mutated - a lot - by the `discoverPodNetworkInterface` method.

Furthermore, it is sometimes used in the `preparePodNetworkInterface` to make local decisions.

Moving it altogether to the podNIC requires changing a lot more than this. I'd rather do that (if at all) on a different PR.

Reply:
Right, so after this PR is merged we will have half of the DHCP logic at podnic and half at the bindings I suppose this is a transitory situation right ? 

Also I would call `generateDhcpConfig` just  `dhcpConfig` since nothing is generated it just an accessor.

Reply:
> Right, so after this PR is merged we will have half of the DHCP logic at podnic and half at the bindings I suppose this is a transitory situation right ?

It's more nuanced that what you're saying; all the dhcp logic would sit in the dhcpConfigurator, but all the dhcp configuration "generation" - filling the ips / routes / etc etc in the `dhcpConfig` member - would remain in the bindings.

I honestly didn't plan to move that away from the bindings, but, I am forced to admit moving the configuration generation also away from what is currently the bindings would improve the code. 

I can agree doing it in a follow-up. Do you want to do it ? 

> 
> Also I would call `generateDhcpConfig` just `dhcpConfig` since nothing is generated it just an accessor.

Sure, will do.

Reply:
ack, let's do that on a follow up.